enricher/standard/src/test/java/io/fabric8/maven/enricher/standard/ImageEnricherTest.java

```diff
@@ -0,0 +1,195 @@
+import io.fabric8.maven.core.config.ResourceConfig;
+import io.fabric8.maven.core.util.KubernetesResourceUtil;
+import io.fabric8.maven.docker.config.ImageConfiguration;
+import io.fabric8.maven.enricher.api.EnricherContext;
+
+import com.jayway.jsonpath.matchers.JsonPathMatchers;
+
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import mockit.Expectations;
+import mockit.Mocked;
+import mockit.integration.junit4.JMockit;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
+/**
+ * @author nicola
+ * @since 14/02/17
+ */
+@RunWith(JMockit.class)
+public class ImageEnricherTest {
+
+    @Mocked
+    private EnricherContext context;
+
+    @Mocked
+    ImageConfiguration imageConfiguration;
+
+    @Before
+    public void prepareMock() {
+        // Setup mock behaviour
+        new Expectations() {{
+            context.getResources(); result = new ResourceConfig.Builder()
+                    .env(Collections.singletonMap("MY_KEY", "MY_VALUE"))
+                    .build();
+
+            imageConfiguration.getName(); result = "busybox";
+            imageConfiguration.getAlias(); result = "busybox";
+            context.getImages(); result = Arrays.asList(imageConfiguration);
+        }};
+    }
+
+    @Test
+    public void checkEnrichDeployment() throws Exception {
+        // Enrich
+        ImageEnricher imageEnricher = new ImageEnricher(context);
+        KubernetesListBuilder builder = new KubernetesListBuilder()
+                .addNewDeploymentItem()
+                .endDeploymentItem();
+
+        imageEnricher.addMissingResources(builder);
+
+        // Validate that the generated resource contains
+        KubernetesList list = builder.build();
+        assertEquals(list.getItems().size(),1);
+
+        String json = KubernetesResourceUtil.toJson(list.getItems().get(0));
+        assertThat(json, JsonPathMatchers.isJson());
+        assertThat(json, JsonPathMatchers.hasJsonPath("$.kind", Matchers.equalTo("Deployment")));
+        assertThat(json, JsonPathMatchers.hasJsonPath("$.spec.template.spec.containers[0].env[0].name", Matchers.equalTo("MY_KEY")));
+        assertThat(json, JsonPathMatchers.hasJsonPath("$.spec.template.spec.containers[0].env[0].value", Matchers.equalTo("MY_VALUE")));
+    }
+
+    @Test
+    public void checkEnrichReplicaSet() throws Exception {
+        // Enrich
+        ImageEnricher imageEnricher = new ImageEnricher(context);
+        KubernetesListBuilder builder = new KubernetesListBuilder()
+                .addNewReplicaSetItem()
+                .endReplicaSetItem();
+
+        imageEnricher.addMissingResources(builder);
+
+        // Validate that the generated resource contains
+        KubernetesList list = builder.build();
+        assertEquals(list.getItems().size(),1);
+
+        String json = KubernetesResourceUtil.toJson(list.getItems().get(0));
```

Comment:
can we extract these assertion into an extra method and use it in the other tests, too ? I'm always a fan of being DRY in tests as well, so that when things changes only a single sport needs to be adapted.